src/ansys/acp/core/_tree_objects/imported_solid_model.py

```diff
@@ -354,8 +354,20 @@ def _create_stub(self) -> imported_solid_model_pb2_grpc.ObjectServiceStub:
         layup_mapping_object_pb2_grpc.ObjectServiceStub,
     )
 
-    def refresh(self, path: _PATH) -> None:
-        """Re-import the solid model from the external file."""
+    def refresh(self, path: _PATH, format: SolidModelImportFormat | None = None) -> None:  # type: ignore
+        """
+        Re-import the solid model from the external file.
+
+        Parameters
+        ----------
+        path :
+            Path of the new input file.
+        format :
+            Switch format of the input file. Optional, uses the current format of the
+            imported solid model if not specified.
+        """
+        if format is not None:
+            self.format = format
```

Comment:
@roosre if it's simple to do, maybe we could add a test case which switches the format on refresh.